benchmark/benchmark.py

```diff
@@ -0,0 +1,229 @@
+# flake8: noqa: E402
+import argparse
+import bz2
+import csv
+import datetime
+import json
+import os
+import sys
+from io import StringIO
+from pathlib import Path
+
+from matplotlib import pyplot as plt  # type: ignore
+
+SCRIPT_DIR = os.path.dirname(os.path.abspath(__file__))
+sys.path.append(os.path.dirname(SCRIPT_DIR))
+
+from eyecite import clean_text, get_citations
+
+csv.field_size_limit(sys.maxsize)
+
+root = Path(__file__).parent.absolute()
+
+MAX_ROWS_IN_MD = 51
+
+
+class Benchmark(object):
+    """Benchmark the different eyecite branches"""
+
+    def get_filepath(self, filename):
+        return Path.joinpath(root, filename)
+
+    def generate_branch_report(self, branch: str) -> None:
+        """Generate Report
+
+        :param branch: Is a branch from main or not
+        :return: None
+        """
+        zipfile = bz2.BZ2File(self.get_filepath("bulk-file.csv.bz2"))
+        csv_data = csv.DictReader(
+            StringIO(zipfile.read().decode()), delimiter=","
+        )
+        count = 0
+        now = datetime.datetime.now()
+        data = []
+        for row in csv_data:
+            id = row.pop("id")
+            newrow = {k: v for k, v in row.items() if v}
+            found_citations = get_citations(
+                clean_text(
+                    list(newrow.values())[0], ["html", "inline_whitespace"]
```

Comment:
This is doing some magic to get the first column in the row that has HTML, right? I think it'd be better to be explicit. Try the field you want with early exit on `or` statements, say:

```
text = row['html'] or row['html_lawbox'] or row['plaintext'] or...
```

That spares you from:
 - popping the ID up above
 - filtering out blank columns up above
 - creating newrow
 - casting newrow.values() to a list
 - grabbing the first non-empty value of the list (in a way that's a bit obscure)

Reply:
its just getting the first non empty opinion.

Reply:
this is much nicer